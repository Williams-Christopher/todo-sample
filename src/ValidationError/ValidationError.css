.todolist_form_error {
    width: 80%;
    border: 1px solid red;
    margin: 10px auto;
}

.todolist_form_error_list {
    list-style: none;
    margin: 0;
    padding: 0;
}