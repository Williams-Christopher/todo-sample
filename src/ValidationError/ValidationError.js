import React from 'react';
import PropTypes from 'prop-types';
import './ValidationError.css';

function ValidationError(props) {
    if (props.hasError && props.message) {
        return (
            <p className='todolist_form_error'>{props.message}</p>
        )
    } else {
        return null;
    }
}

ValidationError.defaultProps = {
    hasError: false,
    message: '',
}

ValidationError.propTypes = {
    hasError: PropTypes.bool,
    message: PropTypes.string,
}

export default ValidationError;