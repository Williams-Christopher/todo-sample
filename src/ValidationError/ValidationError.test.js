import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ValidationError from './ValidationError';

const testMessages = {
    testkey1: 'Message 1',
    testkey2: 'Message 2'
};

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ValidationError />, div);
    ReactDOM.unmountComponentAtNode(div);
})

it('renders with props', () => {
    const tree = renderer.create(<ValidationError errorMessages={testMessages} />).toJSON();
    expect(tree).toMatchSnapshot();
})