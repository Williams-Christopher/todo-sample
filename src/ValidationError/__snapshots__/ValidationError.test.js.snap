// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with props 1`] = `
<aside
  className="todolist_form_error"
>
  <ul
    className="todolist_form_error_list"
  >
    <li>
      Message 1
    </li>
    <li>
      Message 2
    </li>
  </ul>
</aside>
`;
